packages/core/snapshot/write.go

```diff
@@ -158,26 +126,124 @@ func NewLedgerUTXOStatesProducer(nmgr *notarization.Manager) UTXOStatesProducerF
 
 // NewEpochDiffsProducer returns a OutputWithMetadataProducerFunc that provide OutputWithMetadatas from the ledger.
 func NewEpochDiffsProducer(fullEpochIndex, latestCommitableEpoch epoch.Index, nmgr *notarization.Manager) EpochDiffProducerFunc {
-	epochDiffs, err := nmgr.SnapshotEpochDiffs(fullEpochIndex, latestCommitableEpoch)
+	prodChan := make(chan *ledger.EpochDiff)
+	stopChan := make(chan struct{})
+	nmgr.SnapshotEpochDiffs(fullEpochIndex, latestCommitableEpoch, prodChan, stopChan)
 
-	return func() (map[epoch.Index]*ledger.EpochDiff, error) {
-		return epochDiffs, err
+	return func() *ledger.EpochDiff {
+		select {
+		case obj := <-prodChan:
+			return obj
+		case <-stopChan:
+			close(prodChan)
+			return nil
+		}
 	}
 }
 
+func writeEpochDiffs(writeSeeker io.WriteSeeker, diffs *ledger.EpochDiff) error {
+	writeFunc := func(name string, value any) error {
+		return writeFunc(writeSeeker, name, value)
+	}
+
+	spentLen := len(diffs.Spent())
+	if err := writeFunc("epochDiffs spent Len", int64(spentLen)); err != nil {
+		return err
+	}
+
+	s := diffs.Spent()
+	var end int
+	for i := 0; i < spentLen; {
+		if i+chunkSize > spentLen {
+			end = spentLen
+		} else {
+			end = i + chunkSize
```

Comment:
If you are let's say writing 300 outputs, with a chunksize of 100, this is not what you want!
Maybe something of this sort?
```
	for batchNum := 0; batchNum <= len(blockIDs)/m.blockBatchSize; batchNum++ {
		blocks := make([]*tangleold.Block, 0)
		for i := batchNum * m.blockBatchSize; i < len(blockIDs) && i < (batchNum+1)*m.blockBatchSize; i++ {
```

Reply:
the example with 300 should work:
chunk 1: [0:100] (index 0 ~ index 99)
chunk 2: [100:200] (index 100 ~ index 199)
chunk 3: [200:300] (index 200~index 299)

`end` is assigned to `i` in each round